packages/core/src/Serialization/TransactionBody/Value.ts

```diff
@@ -0,0 +1,190 @@
+import * as Cardano from '../../Cardano';
+import * as Crypto from '@cardano-sdk/crypto';
+import { CborReader, CborReaderState, CborWriter } from '../CBOR';
+import { HexBlob, InvalidArgumentError } from '@cardano-sdk/util';
+import { multiAssetsToTokenMap, sortCanonically, tokenMapToMultiAsset } from './Utils';
+
+const VALUE_ARRAY_SIZE = 2;
+
+/**
+ * A Value object encapsulates the quantity of assets of different types,
+ * including ADA (Cardano's native cryptocurrency) expressed in lovelace,
+ * where 1 ADA = 1,000,000 lovelace, and other native tokens. Each key in the
+ * tokens object is a unique identifier for an asset, and the corresponding
+ * value is the quantity of that asset.
+ */
+export class Value {
+  #coin = 0n;
+  #multiasset: Cardano.TokenMap | undefined = undefined;
```

Comment:
Yeah OK, it makes sense in retrospect since we're implementing the CDDL. Our core types smooth out these rough edges we have with the model, so you can ignore the change requests of this nature.